File: lista3/F/f.py
def countMoves():
  n = int(input())
  numbers = set()

  for i in input().split():
    numbers.add(int(i))

  min = 0

  while numbers:
    maxNum = max(numbers)
    numbers.remove(maxNum)
    if maxNum % 2 == 0:
      numbers.add(maxNum // 2)
      min += 1
  
  print(min)

t = int(input())

for i in range(t):
  countMoves()

File: lista3/F/f.cpp
#include <iostream>
#include <algorithm>
#include <set>
using namespace std;

void readSet(int n);
void countMoves();

set<int> s;

int main() { 
  int t;
  cin >> t;
  
  for(int i = 0; i < t; i++) {
    countMoves();
    s.clear();
  }

  return 0;
}

void readSet(int n) {
  int x;
  for(int i = 0; i < n; i++) {
    cin >> x;
    s.insert(x);
  }
}

void countMoves() {
  int n, min = 0;
  cin >> n;
  readSet(n);

  while(!s.empty()) {
    int maxNum = *s.rbegin();
    s.erase(maxNum);
    if(maxNum % 2 == 0) {
      s.insert(maxNum / 2);
      min++;
    }
  }

  cout << min << endl;
}
